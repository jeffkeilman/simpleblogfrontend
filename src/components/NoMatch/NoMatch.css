.NoMatch p {
  text-align: center;
  margin-bottom: 0;
  padding: 0 20px;
}

.NoMatch p:not(.HomeLink) {
  font-size: 3em;
  margin-top: 100px;
}

.NoMatch a {
  color: #000000;
  text-decoration: none;
}

.NoMatch a:hover {
  text-decoration: underline;
}

.HomeLink {
  font-size: 2em;
  margin-top: 0;
}
