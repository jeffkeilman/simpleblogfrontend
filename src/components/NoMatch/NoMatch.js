import './NoMatch.css'
import { Link } from 'react-router-dom'

function NoMatch () {
  return (
    <section className='NoMatch'>
      <p>404! We didn't find anything at that link!</p>
      <p className='HomeLink'><Link to='/'>Take me home!</Link></p>
    </section>
  )
}

export default NoMatch
