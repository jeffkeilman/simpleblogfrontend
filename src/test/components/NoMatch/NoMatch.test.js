import React from 'react'
import renderer from 'react-test-renderer'
import NoMatch from '../../../components/NoMatch/NoMatch'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

describe('NoMatch', () => {
  // snapshot test
  it('renders correctly', () => {
    const nomatch = renderer.create(
      <MemoryRouter>
        <NoMatch />
      </MemoryRouter>
    ).toJSON()
    expect(nomatch).toMatchSnapshot()
  })

  it('renders the 404 copy', () => {
    const { queryByText } = render(
      <MemoryRouter>
        <NoMatch />
      </MemoryRouter>
    )
    expect(queryByText(/404! We didn't find anything at that link!/i)).toBeInTheDocument()
  })

  it('renders the home link', () => {
    const { queryByText } = render(
      <MemoryRouter>
        <NoMatch />
      </MemoryRouter>
    )
    expect(queryByText(/Take me home!/i)).toBeInTheDocument()
  })
})
