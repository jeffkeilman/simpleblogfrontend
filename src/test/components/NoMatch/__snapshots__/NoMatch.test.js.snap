// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch renders correctly 1`] = `
<section
  className="NoMatch"
>
  <p>
    404! We didn't find anything at that link!
  </p>
  <p
    className="HomeLink"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Take me home!
    </a>
  </p>
</section>
`;
